### PR TITLE
chore: overhauled ESLint config

### LINKED ISSUE #312: ESLint complaints in local development
I see there's an `.eslintrc.json` checked in, but a lot of red squigglies exist in my editor. The culprit is the `lint` script missing `src/*.ts`:

```js
- "lint": "eslint src/**/*.ts",
+ "lint": "eslint src/**/*.ts src/*.ts",
```

Would you like a PR that fixes all of the violations? 

```plaintext
 $ npm run lint

> typescript-action@0.0.0 lint
> eslint src/**/*.ts src/*.ts


/Users/josh/repos/pr-compliance-action/src/checks.ts
  27:9   error  'errors' is never reassigned. Use 'const' instead  prefer-const
  37:34  error  Expected '===' and instead saw '=='                eqeqeq

/Users/josh/repos/pr-compliance-action/src/main.ts
   32:7   error  'requiredIssueLabelsAutoClose' is assigned a value but never used     @typescript-eslint/no-unused-vars
   35:7   error  'requiredIssueLabelsComment' is assigned a value but never used       @typescript-eslint/no-unused-vars
  110:63  error  Expected '===' and instead saw '=='                
...
```

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## What type of PR is this? (check all applicable)

- [x] ♻️ Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

Resets the ESLint config by:

1. Having it extend from [typescript-eslint's recommended configurations](https://typescript-eslint.io/linting/configs)
2. Disabling any rules that have complaints which would require non-trivial code changes to fix
3. Fixing remaining violations complaints (most of which were auto-fixable)

Fixes #312 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [x] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Dancing skeleton washing hands in sink with pink soap bubbles

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -188,8 +188,8 @@ async function findExistingReview(pullRequest: {
 }): Promise<PullRequestReview | null> {
   let review
   const {data: reviews} = await client.rest.pulls.listReviews(pullRequest)
-  review = reviews.find(review => {
-    return (review?.user?.login ?? '') == 'github-actions[bot]'
+  review = reviews.find(innerReview => {
+    return (innerReview?.user?.login ?? '') === 'github-actions[bot]'
```
Comment (JoshuaKGoldberg) [inline src/main.ts:192]: https://typescript-eslint.io/rules/no-shadow/

File: .eslintrc.json [hunk 2]
```diff
@@ -1,54 +1,37 @@
...
+    "@typescript-eslint/no-floating-promises": "off",
+    "@typescript-eslint/no-unsafe-argument": "off",
+    "@typescript-eslint/no-unsafe-assignment": "off",
+    "@typescript-eslint/no-unsafe-call": "off",
+    "@typescript-eslint/no-unsafe-member-access": "off",
+    "@typescript-eslint/no-unsafe-return": "off",
+    "@typescript-eslint/restrict-template-expressions": [
+      "error",
+      {"allowAny": true}
+    ],
```
Comment (JoshuaKGoldberg) [inline .eslintrc.json:30]: https://typescript-eslint.io/rules/restrict-template-expressions/#allowany

File: src/checks.ts [hunk 3]
```diff
@@ -3,28 +3,28 @@ import * as conventionalTypes from 'conventional-commit-types'
 import {sync} from 'conventional-commits-parser'
 
 type LintRuleOutcome = {
-  message: String
-  valid: Boolean
+  message: string
+  valid: boolean
```
Comment (JoshuaKGoldberg) [inline src/checks.ts:7]: https://typescript-eslint.io/rules/ban-types/

File: src/main.ts [hunk 4]
```diff
@@ -166,7 +166,7 @@ async function closePullRequest(number: number) {
     state: 'closed'
   })
 }
-async function escapeChecks(checkResult: boolean, message: string) {
+function escapeChecks(checkResult: boolean, message: string) {
```
Comment (JoshuaKGoldberg) [inline src/main.ts:169]: https://typescript-eslint.io/rules/promise-function-async/

**Comment by mtfoley** [comment]
@JoshuaKGoldberg did you already run ‘npm run all’ leading up to the commits? I haven’t checked this locally, I’m just curious. Depending on how the compile step handles changes to *.ts, there should be changes in the “dist” directory.

**Comment by JoshuaKGoldberg** [comment]
Ah sorry, I forgot that was needed in this repo - done!

File: dist/header.hbs [hunk 5]
```diff
@@ -1,8 +1,4 @@
-{{#if isPatch~}}
-  ###
-{{~else~}}
-  ##
-{{~/if}} {{#if @root.linkCompare~}}
+## {{#if @root.linkCompare~}}
```
Comment (JoshuaKGoldberg) [inline dist/header.hbs:1]: 🤷

Comment (mtfoley) [reply dist/header.hbs:1]: I have yet to pay attention to why this repo outputs handlebars files, no worries 😉

**Comment by mtfoley** [comment]
Thanks for the PR!

**Pull request merged**